vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -206,16 +170,53 @@ export class WebRequestWorkerSingleton
         }, this.timeoutMsFromCtx(ctx));
 
         // Make the web request
-        const response = await this.client.get(url, { signal: timeoutCancelTokenHook.signal, ...options });
-        // Timeout for Web Request -> Not Timer. (Don't want to introduce more CPU time into timer)
-        clearTimeout(timeout);
-
-        this.reportTimeAnalytics(response, options, url, ctx);
+        let response;
+        if (this.client !== null)
+        {
+            response = await this.client.get(url, { signal: timeoutCancelTokenHook.signal, ...options });
+            // Timeout for Web Request -> Not Timer. (Don't want to introduce more CPU time into timer)
+            clearTimeout(timeout);
+            this.reportTimeAnalytics(response, options, url, ctx);
+        }
+        else
+        {
+            response = await this.getFetchResponse(url, ctx, useFetchDownload);
+            clearTimeout(timeout);
+        }
 
         // Response
         return response;
     }
 
+    private async getFetchResponse(url: string, ctx: IAcquisitionWorkerContext, returnDownloadStream = false): Promise<any>
```

Comment:
This is work the client is supposed to do, right? I'm a bit confused as to why we're making a temp client rather than just setting this.client to the alt client then making the web request as normal?

Reply:
I hope my comment above clarifies. This isnt a temp client, it is an entirely different web lib (native fetch) vs axios if axios does not work